tests/integration/dns_test.py

```diff
@@ -617,3 +620,44 @@ def test_purge_dns_full_config(static_dns_with_options):
 
     current_state = libnmstate.show()
     assert not current_state[DNS.KEY][DNS.CONFIG]
+
+
+def test_kernel_mode_dns_and_purge():
+    desired_state = load_yaml(
+        """---
+        dns-resolver:
+          config:
+            search:
+            - example.com
+            - example.org
+            server:
+            - 2001:4860:4860::8888
+            - 2001:4860:4860::8844
+            - 8.8.4.4
+            - 8.8.8.8
+            options:
+            - debug
+            - rotate
+        """
+    )
+    with disable_service("NetworkManager"):
```

Comment:
What is the reason that we have to disable NetworkManager before applying the DNS in kernel mode? Will `nmstatectl apply --kernel` not work as expected if not disabling NetworkManager ?

Reply:
Indicate we are not using NetworkManager to apply the network config.

In theory, with NetworkManager started, this kernel mode should also works well. Even not so, it's a bug of NM, not nmstate, hence no test of such use case in nmstate CI.